### ISSUE TITLE
Loss Function Calculation During Training Process in Code

### ISSUE THREAD

**Issue by EthanC20** [NUSTM/MECPE#11]
Hi, thanks for the great work! I have a question about the training logic.

https://github.com/NUSTM/MECPE/blob/359c97c74416aecd6b9e62738db10a37b4aa0104/step1.py#L338

In `step1.py`, when computing `loss_op`, there is a conditional logic for `FLAGS.use_x_a`, where `loss_emo_x_a` is calculated and added. However, there is no corresponding condition or addition for `loss_emo_x_v` from the visual modality. Is this an oversight in the code design, or an intentional asymmetric treatment? Furthermore, I noticed that in the original paper, adding the visual modality led to worse performance in the ablation studies — could this potentially be due to an issue in the loss computation logic?

<img width="1049" alt="Image" src="https://github.com/user-attachments/assets/14e6b9b4-4b90-42ec-a35d-b63e6de59350" />

Would appreciate any clarification on this.

**Comment by ffcarina** [comment]
@EthanC20 Thank you for your interest in our work. The absence of a separate loss for the visual modality is intentional, not an oversight. We found that adding such a loss negatively impacted performance, likely due to the significant amount of noise in the video, which makes the model difficult to capture crucial emotion-related information using general visual features. 
We encourage you to explore alternative optimization approaches.

**Comment by EthanC20** [comment]
Thank you for your prompt response and valuable guidance. I’ve now extracted more fine-grained visual features from the ECF dataset and plan to conduct further experiments by incorporating a loss function specifically for the visual modality. I really appreciate your insights—they’ve been very helpful in shaping the next steps of my research.
